File: app/src/main/java/bj/discogsbrowser/model/listing/Listing.java

package bj.discogsbrowser.model.listing;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import bj.discogsbrowser.model.common.RecyclerViewModel;
import bj.discogsbrowser.utils.DateFormatter;
import bj.discogsbrowser.wrappers.DateUtilsWrapper;
import bj.discogsbrowser.wrappers.SimpleDateFormatWrapper;

public class Listing implements RecyclerViewModel
{
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("price")
    @Expose
    private Price price;
    @SerializedName("allow_offers")
    @Expose
    private Boolean allowOffers;
    @SerializedName("sleeve_condition")
    @Expose
    private String sleeveCondition;
    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("condition")
    @Expose
    private String condition;
    @SerializedName("posted")
    @Expose
    private String posted;
    @SerializedName("ships_from")
    @Expose
    private String shipsFrom;
    @SerializedName("uri")
    @Expose
    private String uri;
    @SerializedName("comments")
    @Expose
    private String comments;
    @SerializedName("seller")
    @Expose
    private Seller seller;
    @SerializedName("shipping_price")
    @Expose
    private ShippingPrice shippingPrice;
    @SerializedName("release")
    @Expose
    private Release release;
    @SerializedName("resource_url")
    @Expose
    private String resourceUrl;
    @SerializedName("audio")
    @Expose
    private Boolean audio;

    public String getStatus()
    {
        return status;
    }

    public void setStatus(String status)
    {
        this.status = status;
    }

    public Price getPrice()
    {
        return price;
    }

    public void setPrice(Price price)
    {
        this.price = price;
    }

    public Boolean getAllowOffers()
    {
        return allowOffers;
    }

    public void setAllowOffers(Boolean allowOffers)
    {
        this.allowOffers = allowOffers;
    }

    public String getSleeveCondition()
    {
        return sleeveCondition;
    }

    public void setSleeveCondition(String sleeveCondition)
    {
        this.sleeveCondition = sleeveCondition;
    }

    public String getId()
    {
        return id;
    }

    public void setId(String id)
    {
        this.id = id;
    }

    public String getCondition()
    {
        return condition;
    }

    public void setCondition(String condition)
    {
        this.condition = condition;
    }

    public String getPosted()
    {
        return posted;
    }

    public void setPosted(String posted)
    {
        this.posted = posted;
    }

    public String getShipsFrom()
    {
        return shipsFrom;
    }

    public void setShipsFrom(String shipsFrom)
    {
        this.shipsFrom = shipsFrom;
    }

    public String getUri()
    {
        return uri;
    }

    public void setUri(String uri)
    {
        this.uri = uri;
    }

    public String getComments()
    {
        return comments;
    }

    public void setComments(String comments)
    {
        this.comments = comments;
    }

    public Seller getSeller()
    {
        return seller;
    }

    public void setSeller(Seller seller)
    {
        this.seller = seller;
    }

    public ShippingPrice getShippingPrice()
    {
        return shippingPrice;
    }

    public void setShippingPrice(ShippingPrice shippingPrice)
    {
        this.shippingPrice = shippingPrice;
    }

    public Release getRelease()
    {
        return release;
    }

    public void setRelease(Release release)
    {
        this.release = release;
    }

    public String getResourceUrl()
    {
        return resourceUrl;
    }

    public void setResourceUrl(String resourceUrl)
    {
        this.resourceUrl = resourceUrl;
    }

    public Boolean getAudio()
    {
        return audio;
    }

    public void setAudio(Boolean audio)
    {
        this.audio = audio;
    }

    @Override
    public String getTitle()
    {
        return release.getDescription();
    }

    @Override
    public String getSubtitle()
    {
        DateFormatter dateFormatter = new DateFormatter(new DateUtilsWrapper(), new SimpleDateFormatWrapper());
        return "Posted: " + dateFormatter.formatIsoDate(posted);
    }

    @Override
    public String getThumb()
    {
        return release.getThumbnail();
    }

    @Override
    public String getType()
    {
        return "listing";
    }
}

File: app/src/main/java/bj/discogsbrowser/model/listing/Listing.kt
package bj.discogsbrowser.model.listing

import bj.discogsbrowser.model.common.Price
import bj.discogsbrowser.model.common.RecyclerViewModel
import bj.discogsbrowser.model.common.Seller
import bj.discogsbrowser.utils.DateFormatter
import bj.discogsbrowser.wrappers.DateUtilsWrapper
import bj.discogsbrowser.wrappers.SimpleDateFormatWrapper
import com.google.gson.annotations.SerializedName

/**
 * Created by Josh Laird on 19/05/2017.
 */
data class Listing(var status: String = "",
                   var price: Price = Price("0", 0.0),
                   @SerializedName("allow_offers") var allowOffers: Boolean = false,
                   @SerializedName("sleeve_condition") var sleeveCondition: String = "",
                   @get:JvmName("getId_") var id: String = "",
                   var condition: String = "",
                   var posted: String = "",
                   @SerializedName("ships_from") var shipsFrom: String = "",
                   var uri: String = "",
                   var comments: String = "",
                   var seller: Seller = Seller(),
                   var shippingPrice: ShippingPrice = ShippingPrice("", ""),
                   var release: ListingRelease = ListingRelease(),
                   var resourceUrl: String = "",
                   var audio: Boolean = false
) : RecyclerViewModel {
    override fun getTitle(): String {
        return release.description
    }

    override fun getSubtitle(): String {
        val dateFormatter = DateFormatter(DateUtilsWrapper(), SimpleDateFormatWrapper())
        return "Posted: " + dateFormatter.formatIsoDate(posted)
    }

    override fun getThumb(): String {
        return release.thumbnail
    }

    override fun getType(): String {
        return "listing"
    }

    override fun getId(): String {
        return id
    }
}